src/rebuild/srv.c

```diff
@@ -1694,78 +1774,91 @@ rebuild_task_complete_schedule(struct rebuild_task *task, struct ds_pool *pool,
+			    pool, task->dst_reclaim_ver - 1, rgt->rgt_stable_epoch,
+			    task->dst_new_layout_version, &task->dst_tgts, RB_OP_FAIL_RECLAIM,
+			    rgt->rgt_stop_admin, delay_sec);
+			DL_CDEBUG(rc, DLOG_ERR, DLOG_INFO, rc,
+				  DF_RB ": errno " DF_RC ", schedule %u(%s)", DP_RB_RGT(rgt),
+				  DP_RC(rgt->rgt_status.rs_errno), RB_OP_FAIL_RECLAIM,
+				  RB_OP_STR(RB_OP_FAIL_RECLAIM));
 		}
 
 		/* Then check if it needs to retry */
-		retry_rebuild_task(task, rgt->rgt_status.rs_errno, &retry_opc);
+		retry_rebuild_task(task, rgt, &retry_opc);
 		if (retry_opc == RB_OP_NONE)
 			D_GOTO(complete, rc);
 
 		rc = ds_rebuild_schedule(pool, task->dst_map_ver, rgt->rgt_stable_epoch,
-					 task->dst_new_layout_version, &task->dst_tgts,
-					 retry_opc, 5);
-		DL_INFO(rc, DF_UUID" opc %u/%u, error %d, re-scheduled opc %u.",
-			DP_UUID(task->dst_pool_uuid), task->dst_rebuild_op, task->dst_map_ver,
-			rgt->rgt_status.rs_errno, retry_opc);
+					 task->dst_new_layout_version, &task->dst_tgts, retry_opc,
+					 false /* stop_admin */, delay_sec);
+		DL_CDEBUG(rc, DLOG_ERR, DLOG_INFO, rc, DF_RB ": errno " DF_RC ", schedule retry",
+			  DP_RB_RGT(rgt), DP_RC(rgt->rgt_status.rs_errno));
 	} else if (task->dst_rebuild_op == RB_OP_REBUILD || task->dst_rebuild_op == RB_OP_UPGRADE) {
-		/* Otherwise schedule reclaim for reintegrate/extend/upgrade. */
+		/* Schedule reclaim for successful op:Rebuild
+		 * (exclude/drain/reintegrate/extend/upgrade). */
 		rgt->rgt_status.rs_state = DRS_IN_PROGRESS;
 
-		D_DEBUG(DB_REBUILD, DF_UUID" opc %u/%u, error %d, schedule RECLAIM.\n",
-			DP_UUID(task->dst_pool_uuid), task->dst_rebuild_op, task->dst_map_ver,
-			rgt->rgt_status.rs_errno);
 		rc = ds_rebuild_schedule(pool, task->dst_map_ver, rgt->rgt_reclaim_epoch,
-					 task->dst_new_layout_version,
-					 &task->dst_tgts, RB_OP_RECLAIM, 5);
-		if (rc != 0)
-			D_ERROR("reschedule reclaim, "DF_UUID" failed: "DF_RC"\n",
-				DP_UUID(task->dst_pool_uuid), DP_RC(rc));
+					 task->dst_new_layout_version, &task->dst_tgts,
+					 RB_OP_RECLAIM, false /* stop_admin */, delay_sec);
+		DL_CDEBUG(rc, DLOG_ERR, DLOG_INFO, rc, DF_RB ": errno " DF_RC ", schedule %u(%s)",
+			  DP_RB_RGT(rgt), DP_RC(rgt->rgt_status.rs_errno), RB_OP_RECLAIM,
+			  RB_OP_STR(RB_OP_RECLAIM));
 	}
 
 complete:
+	/* Update the rebuild complete status for pool query */
 	if (task->dst_rebuild_op != RB_OP_FAIL_RECLAIM) {
-		/* Update the rebuild complete status for pool query */
-		D_DEBUG(DB_REBUILD, DF_UUID" opc %u/%u, state %d error %d, update status.\n",
-			DP_UUID(task->dst_pool_uuid), task->dst_rebuild_op, task->dst_map_ver,
-			rgt->rgt_status.rs_state, rgt->rgt_status.rs_errno);
+		/* During and after an admin stop, pool query should show an indication.
+		 * Query rs_state from original Rebuild = IN_PROGRESS until Fail_reclaim finishes
+		 * later.
+		 */
+		if (rgt->rgt_stop_admin && rgt->rgt_abort)
+			rgt->rgt_status.rs_errno = -DER_OP_CANCELED;
+		update_rb_status = true;
```

Comment:
what's the effect of "update_rb_status" flag? seems only will affect the L1854's log? ( the return code of this func is ignored).

Reply:
removing this boolean and replicating the DL_CLDEBUG() to be immediately after the status update calls.